DMTN-165.tex

```diff
@@ -50,7 +56,289 @@
+\citeds{DMTN-093}.
+As the Alert Production pipeline runs, it would produce both the full-sized alert packets and their lightweight counterparts, both in Avro format.
+The lightweight alerts would be distributed to community brokers and science users using Kafka.
+The full-sized alerts would be stored on disk in an object store or other system accessible by HTTP.
+Brokers and users would obtain an identifier from lightweight alerts and request the corresponding full-sized alerts via HTTP.
+
+% more about the rate limit implementation?
+
+%  URL for packet retrieval -- longevity
+The link to retrieve the full-sized alert must be permanent, even if the full-sized alert is moved.  
+This suggests use of an identifier such as a DOI, URI, or IVORN\footnote{\url{https://www.ivoa.net/documents/IVOAIdentifiers/20160523/index.html}}.
+
+\section{Implications for the Alert Filtering Service} \label{sec:alertfiltering}
+
+In the current baseline, Rubin Observatory runs an Alert Filtering Service (AFS) within the Data Facility that allows a limited number of users ($\geq$100 simultaneous\dmreq{0343}\reqparam{numBrokerUsers}) with Data Rights to upload filters and have a limited number of alerts ($\leq$ 20 per visit\reqparam{numBrokerAlerts}) forwarded to them.
+In addition to the limited capacity, the AFS restricts the user filters to only operate on the contents of the alert packets themselves.
+There is currently no latency requirement for alert delivery through the AFS.
+
+We suggest that the lightweight alert packet design would obviate the scientific need for the AFS, reducing Project scope.
+Instead of providing filter code to run on Data Facility services, users with data rights could run filters on their own hardware with a great deal more flexibility.
+In particular, users would be free to build filters that crossmatched to external catalogs or realtime streams or performed computationally-intensive fitting or image-processing tasks.
+They would be free to choose their own programming language, libraries, and environment.
+While the need to implement their own local service would impose additional difficulty relative to a Project-provided AFS, the Project could provide simple tutorial material for a basic implementation.
+
+%\textbf{estimate data rates for 100x20.  20 full-sized alerts per filter had to come out anyway}
+
+One particular advantage of this scheme to users is that within their overall rate limit budget, they could receive more alerts from single visits at cost of higher latency.
+The AFS is likely to have had hard limits at 20 full-sized alerts per visit, after which no more would be forwarded.
+In contrast, a user wanting many more alerts from a high-priority single visit (say, a Deep Drilling field) could simply request more full-sized alerts over a longer period of time.
+
+Descoping the AFS would reduce on-Project construction and operations effort.
```

Comment:
I think the alert filtering system is a pretty big technical risk. Descoping it is a big feature of the proposal to me.